libsolidity/ast/AST.cpp

```diff
@@ -319,6 +319,37 @@ FunctionDefinitionAnnotation& FunctionDefinition::annotation() const
 	return initAnnotation<FunctionDefinitionAnnotation>();
 }
 
+FunctionDefinition const& FunctionDefinition::resolveVirtual(
+	ContractDefinition const& _mostDerivedContract,
+	ContractDefinition const* _searchStart
+) const
+{
+	solAssert(!isConstructor(), "");
+	if (_searchStart == nullptr && !virtualSemantics())
+		return *this;
+
+	if (auto s = dynamic_cast<ContractDefinition const*>(scope()))
```

Comment:
When is the scope of a function definition not a contract?

Reply:
right!